src/palace/manager/api/circulation.py

```diff
@@ -869,6 +869,62 @@ def sync_patron_activity(self, patron: Patron, pin: str | None) -> None:
 
         self.sync_loans(patron, remote_loans, local_loans)
         self.sync_holds(patron, remote_holds, local_holds)
+        self.delete_loans_and_holds_no_longer_available(patron)
```

Comment:
Minor: This is fine, and it can go in as is, but I think it would be a better user experience, if this ran as a scheduled reaper task, rather then tacking it on to sync patron activity.

Patron activity sync is triggered, _after_ a patron requests their loans feed, since its slow to pull information from every distributor, and we assume our system is mostly up to date. So the way this is implemented, patrons will still see their outdated loans / holds when they first look at them, but upon refresh they will be gone. 

Since we have the information for all this locally, instead of waiting for the patron to request the feed, we could occasionally be doing this in the background.

Reply:
@jonathangreen :  a couple of questions here
1) Frequency for the the periodic background update?  I'm imagining that every 12 hours would probably be more sufficiently frequent.  What do you think? 
2) If I add a scheduled background task, we want to move `delete_loans_and_holds_no_longer_available` rather than invoke it both places (ie in the background task as well as the patron sync)

Reply:
Yes, I think removing it from patron sync makes sense.